Source/ARTClientOptions.h

```diff
@@ -16,6 +16,9 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/**
+ ARTClientOptions is used in the AblyRealtime constructor’s options argument.
```

Comment:
This problem begs automation =)

Reply:
Is linking even possible prior to Xcode 13? I know there was something like a `tag:` but it links to method in the file where `tag:` is declared. More info https://stackoverflow.com/a/54564301

Reply:
@lukasz-szyszkowski thanks for useful link. I know tools exist for exporting docs from source code. But having different platforms this is not very useful. What would be useful, is having universal docs in programmatically accessed form (xml or json), so programmers could "suck" it back into their sources (by hands or script). WDYT? Brief googling haven't shown anything promising.

Reply:
Do you mean something opposite to Jazzy (generating doc from source code)? 
  
  For example:
  - there is serialized documentation somewhere on the web
  - the parts of the code that need to be documented are tagged somehow (to determine)
  - on `framework build` some script is fetching doc from the web and replacing tagged place with description

?

Reply:
Yes, exactly.

Reply:
Sounds good to me, but this could be something more than just a simple script.
The question is: does tags are replaced in source files or script is creating "a copy" of the codebase before building? And what about cocoapods and SPM then?
IMO working on source files is easier but this solution could generate a lot of git changes, or it'll be smart enough to not change everything every time.

To summarize, this could be a nice side project not only for cocoa but for other platforms too. It needs to be talked about with @QuintinWillison

Reply:
As I see it, one could just write something like this before class/method definition:
/***
  @autodoc <class/method reference>
*/
And script will replace it with actual description. At first it can be done locally on a programmer's machine, so git would not be affected directly. But later it could be done as a github workflow f.e.

Reply:
I've explored this topic area in the past (at my last employer, where we also looked at SDKs) but it's never going to be as simple as a simple placeholder replace. There are many factors to consider, as well as stakeholders / users.

Good quality API reference documentation should be self-referential - e.g. linking to referenced properties, methods, arguments, etc.. The idioms vary too dramatically in terms of APIs to make this something that can be entirely driven automatically.

I've mused in the past over whether it's possible to perhaps look at it from another angle and perhaps treat doc comments with some kind of bespoke lint tool that tries to work out whether the non-idiom-specific pieces of the content conform with some central source of proof... but that'll have a tonne of edge cases and engineering effort involved also.

Discussing this in more depth under this ticket is perhaps not the right place - however this is something we need to discuss as a wider team sooner than later so I'll tag @AndyNicks, @marklewin and @tbedford for visibility.

Reply:
I've established a venue where this conversation can continue, for all SDKs: https://github.com/ably/docs/issues/1201